extensions/socket/internal.m

```diff
@@ -0,0 +1,362 @@
+#import <LuaSkin/LuaSkin.h>
+#import <CocoaAsyncSocket/GCDAsyncSocket.h>
+
+static const char *USERDATA_TAG = "hs.socket";
+
+int refTable;
+
+@interface HSAsyncSocket : GCDAsyncSocket
+@property int callback;
+@end
+
+
+static void callback(HSAsyncSocket *asyncSocket, NSData *data) {
+    LuaSkin *skin = [LuaSkin shared];
+    NSString *utf8Data = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+
+    if (!asyncSocket.callback) {
+        lua_getglobal(skin.L, "print");
```

Comment:
Is this more of a debugging choice? Normally we'd throw an error at not having a callback defined when we're getting system callback data